### ISSUE TITLE
state_output_profile defaults are confusing

### ISSUE THREAD

**Issue by bobrik** [saltstack/salt#35166]
### Description of Issue/Question

Defaulting to different values is confusing:

```
$ fgrep 'state_output_profile' -r salt
salt/output/highstate.py:            if __opts__.get('state_output_profile', True):
salt/output/highstate.py:        if __opts__.get('state_output_profile', False):
salt/output/highstate.py:        if __opts__.get('state_output_profile', False):
salt/output/highstate.py:        if __opts__.get('state_output_profile', False):
```

Especially if in the majority of cases default is not what is documented:

```
$ fgrep '#state_output_profile' -r conf
conf/minion:#state_output_profile: True
conf/proxy:#state_output_profile: True
```

### Versions Report

```
Salt Version:
           Salt: 2016.3.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: Not Installed
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: 0.3.6
         PyYAML: 3.11
          PyZMQ: Not Installed
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: Not Installed

System Versions:
           dist: debian 8.5
        machine: x86_64
        release: 4.4.14-cloudflare
         system: Linux
        version: debian 8.5
```

#29217 is related

cc @clan

**Comment by Ch3LL** [comment]
@bobrik are you running into a specific problem with `state_output_profile`? Is the usage confusing? Can we improve the docs? I'm a little unclear what the issue at hand is exactly. Any other clarifying information would be appreciated. Thanks.

**Comment by bobrik** [comment]
@Ch3LL I think that default in code should be the same as in example config. I am happy to make a PR if salt developers agree.

If salt developers disagree, I'd like to understand why.

**Comment by cachedout** [comment]
Closed via #35271
